File: openai-api/whisper-transcribe.py
from openai import OpenAI
from os import environ

oaikey = open("/tmp/mycompany_openai.txt", "r").read().strip()
environ["OPENAI_API_KEY"] = oaikey

client = OpenAI()

audio_file = open("a.mp3", "rb")
transcript = client.audio.transcriptions.create(
  model="whisper-1",
  file=audio_file
)

print(transcript)

File: openai-api/whisper-transcribe.js
const fs = require('fs');
const OpenAI = require('openai');

const oaikey = fs.readFileSync("/tmp/oaikey.txt").toString().trim();
const openai = new OpenAI({apiKey: oaikey});

async function main() {
  const transcription = await openai.audio.transcriptions.create({
    file: fs.createReadStream("a.mp3"),
    model: "whisper-1",
  });
	console.log(transcription);
  console.log(transcription.text);
}
main();
